src/controller.rs

```diff
@@ -40,6 +40,11 @@ impl Controller {
         // Start the dummy adapter.
         let dummy_adapter = DummyAdapter::new(self.sender.clone(), self.context.clone());
         dummy_adapter.start();
+
+        {
+            let mut context = self.context.lock().unwrap();
+            context.start_tunnel().unwrap();
+        }
```

Comment:
I see. I had to do something just like this over here:
https://github.com/fxbox/foxbox/pull/59/files#diff-e1ea9eba222b6cc49837b3dd765989dfR21

In this case, since `context` is about to go out of scope, maybe a comment would suffice instead?